### ISSUE TITLE
Issue when querying several times the same custom scalar field

### ISSUE THREAD

**Issue by eole1712** [apollographql/router#1583]
Hi guys

**Describe the bug**
When in a query we ask several times for the same field (without using aliases) that contains a non-nullable and non-empty array of scalars (such as UUID), the query answer becomes null. 

**To Reproduce**
1. Setup services
Setup a service with the followed schema :
```
scalar UUID

type User {
  id: Int!
  name: String!
  tag_ids: [UUID!]!
}

type Query {
  user: User
}
```

2. Submit request

This request will work
```
query {
  user {
    name
    tag_ids
  }
}
```

but this one will return `user: null`
```
query {
  user {
    name
    tag_ids
    tag_ids
  }
}
```

the generated subgraph request for the two examples is :
```
query variant__servicea__0{user{name tag_ids}}

```
and the subgraph is responding the same answers for both, however I think there is a post-processing problem afterwards

**Expected behavior**
We expect the router to return the same content, no matter how many times we have requested the same key

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] MacBook Pro (14-inch, 2021) M1 Pro / MacOS 12.1
 - Version [e.g. 22] Router v0.16.0

**Comment by Geal** [comment]
hi! thanks for the report. I think this is related to response formatting, we will take a look

**Comment by eole1712** [comment]
Hello @Geal, thanks for the quick answer

I created a PR #1585 to fix the issue and added a test to cover it. I hope it helps you. :)

Have a good day 💛

**Comment by Geal** [comment]
fixed in #1585

Thanks @eole1712 !
